### PR TITLE
An error when loading schema at tablet start up shouldn't be fatal

### PR DESCRIPTION
**Pull request by aaijazi**
@sougou

### CODE REVIEW

**Comment by aaijazi** [comment]
Ready for review now, has all the tests in place

File: go/vt/tabletserver/schema_info_test.go [hunk 1]
```diff
@@ -288,19 +288,20 @@ func TestSchemaInfoCreateOrUpdateTableFailedDuetoExecErr(t *testing.T) {
...
-		"CreateOrUpdateTable should fail because it could not tables from MySQL",
-		ErrFail,
-	)
 	schemaInfo.Open(&appParams, &dbaParams, getSchemaInfoTestSchemaOverride(), false)
 	defer schemaInfo.Close()
+	// should silently fail: no errors returned, but increment a counter
 	schemaInfo.CreateOrUpdateTable(context.Background(), "test_table")
+	internalErrorCounts := schemaInfo.queryServiceStats.InternalErrors.Counts()
+	schemaErrors, ok := internalErrorCounts["Schema"]
+	if !ok || schemaErrors != 1 {
```
Comment (sougou) [inline go/vt/tabletserver/schema_info_test.go]: You may want to do a before and after diff instead, just to be safe.

**Comment by sougou** [comment]
LGTM. One minor comment.

File: go/vt/tabletserver/schema_info_test.go (hunk 1 above)
Comment (aaijazi) [inline go/vt/tabletserver/schema_info_test.go]: Done.

**Comment by aaijazi** [comment]
Going to merge optimistically (the only tests that I modified in the latest commit all pass locally), I'll revert if tests fail.

**Comment by aaijazi** [comment]
For the record, the tests failed locally due to a flake in initial_sharding_bytes, but successfully passed on master.

**Pull request merged**